### PR TITLE
Upgrade SignalR.Core to netcoreapp3.0

### PR DESCRIPTION
**Pull request by natemcmaster**
This updates samples, tests, and projects to netcoreapp3.0. 

Notably, the SignalR client remains netstandard2.0

Part of https://github.com/aspnet/AspNetCore/issues/3754

### CODE REVIEW

File: benchmarkapps/BenchmarkServer/BenchmarkServer.csproj [hunk 1]
```diff
@@ -23,8 +21,8 @@
 
   <!-- These references are used when running on the Benchmarks Server -->
   <ItemGroup Condition="'$(BenchmarksTargetFramework)' != ''">
-    <PackageReference Include="Microsoft.AspNetCore.All" Version="$(MicrosoftAspNetCoreAllPackageVersion)" />
```
Comment (BrennanConroy) [inline benchmarkapps/BenchmarkServer/BenchmarkServer.csproj]: Hmm, don't change this stuff unless you know 100% that the benchmarks infrastructure works with it

File: Directory.Build.targets [hunk 2]
```diff
@@ -1,7 +1,2 @@
 ﻿<Project>
   <PropertyGroup>
-    <RuntimeFrameworkVersion Condition=" '$(TargetFramework)' == 'netcoreapp2.2' ">$(MicrosoftNETCoreApp22PackageVersion)</RuntimeFrameworkVersion>
-    <!-- aspnet/BuildTools#662 Don't police what version of NetCoreApp we use -->
-    <NETCoreAppMaximumVersion>99.9</NETCoreAppMaximumVersion>
-  </PropertyGroup>
-</Project>
```
Comment (BrennanConroy) [inline Directory.Build.targets]: Looks like you deleted too much or not enough

File: samples/ClientSample/ClientSample.csproj [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netcoreapp2.2;net461</TargetFrameworks>
+    <TargetFrameworks>netcoreapp3.0;net461</TargetFrameworks>
```
Comment (BrennanConroy) [inline samples/ClientSample/ClientSample.csproj]: Remove fullframework?

File: benchmarkapps/BenchmarkServer/BenchmarkServer.csproj (hunk 1 above)
Comment (natemcmaster) [reply benchmarkapps/BenchmarkServer/BenchmarkServer.csproj]: We killed Microsoft.AspNetCore.All. Benchmarks infrastructure will have to update.

File: samples/ClientSample/ClientSample.csproj (hunk 3 above)
Comment (natemcmaster) [reply samples/ClientSample/ClientSample.csproj]: The client still supports it because the client is netstandard.

**Comment by natemcmaster** [comment]
Tests passing now :)

**Pull request merged**